### PR TITLE
Null the collection immediately

### PR DESCRIPTION
**Pull request by eswordthecat**
Also add the two procs from tg saycodee fixes.

### CODE REVIEW

**Comment by ComicIronic** [comment]
MERGE MERGE MERGE

**Comment by eswordthecat** [comment]
It will be automagically stripped at the upstream commits.
ALSO MERGE MERGE MERGE

**Comment by ComicIronic** [comment]
I'm not worried about the history, I'm worried about why you needed 3 merges to just start coding.

File: code/game/say.dm [hunk 1]
```diff
@@ -127,6 +127,18 @@ var/list/freqtoname = list(
 		return returntext
 	return "[copytext("[freq]", 1, 4)].[copytext("[freq]", 4, 5)]"
 
+/proc/attach_spans(input, list/spans)
```
Comment (ComicIronic) [inline code/game/say.dm]: What's the use of this proc?

File: code/game/say.dm [hunk 2]
```diff
@@ -127,6 +127,18 @@ var/list/freqtoname = list(
...
+	return "[message_spans(spans)][input]</span>"
+
+/proc/message_spans(list/spans)
+	var/output = "<SPAN CLASS='"
+
+	for(var/span in spans)
+		output = "[output][span] "
```
Comment (ComicIronic) [inline code/game/say.dm]: Does putting all the spans together actually work?
<span class='dangersanssinister'>

File: code/game/say.dm (hunk 1 above)
Comment (eswordthecat) [inline code/game/say.dm]: For another saycode pr

File: code/game/say.dm (hunk 2 above)
Comment (eswordthecat) [inline code/game/say.dm]: http://stackoverflow.com/questions/5482677/how-to-apply-two-css-classes-to-a-single-div-span/5482693#5482693

**Comment by eswordthecat** [comment]
I'm just used to check updates at upstream.

**Pull request merged**